notebooks/Notebook3.md

```diff
@@ -2,33 +2,56 @@
+
+“One note I would have is that your StringGrouping.java file contains if(groupsMatcher.matches()) which works to prevent errors but it seems like nothing happens if a user inputs a string that doesn't match the regex pattern so I would recommend adding an else case which lets users know they've made an error”
+
+
+“The "run" play could also be referred to as a "rush"”
+
+
+I will address the previous comments from last week. Firstly, I hope you don’t mind the rant but most of this week's notebook will consist of the many many errors I ran into. Secondly, I went into office hours and prof Ben was able to provide me with a layout that has the capabilities of running a java app with java fx. This really helped this week and I appreciate the help from him. Thirdly, to address the set vs add. I can see how just looking over the code you may find that these functions are more related to an add feature function. However in my mind they are more of a set/update function. For the time being I will keep them as a set function title but might change them later. Also they do not really affect the user so like you said it is not at the top of my list. Fourthly, This week I added in error handling so the part of the code if(groupsMatcher.matches()) is actually more relevant now. Also when I worked with regex last summer during my internship that is how we used it so its kinda a habit. Lastly, wow I cannot believe I did not think to use rush. I added that in Thanks!!
+
 
 ## Description
 
-**TODO:** Fill in this part with information about your work this week:
-important design decisions, changes to previous decisions, open questions,
-exciting milestones, preliminary results, etc. Feel free to include images
-(e.g., a sketch of the design or a screenshot of a running program), links to
-code, and any other resources that you think will help clearly convey your
-design process.
+This week was a lot of setup with codespaces and dependencies however I was glad to get some stuff done with error handling and setting up the java.app on codespaces. Moreover, I struggled a ton with trying to get an API working.
+
+To start the week I began by adding in error handling. The user now has a workable interface that allows the user to know when the language that they used is not up to par. If there are parts of their input that are not correct they will receive a message stating that _____ part of their input was wrong and also a provided solution for it. The solution aspect is very similar to what I will provide as documentation. ([link to interface](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/my-dsl/app/src/main/java/my/dsl/Interface/App.java)) ([link to error message files](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/tree/main/my-dsl/app/src/main/java/my/dsl/Interface)) ([link to example error handling](https://github.com/hmc-cs111-spring2023/Bbecker8-Artifact/blob/main/my-dsl/app/src/main/java/my/dsl/Parser/StringGrouping.java)).
```

Comment:
I looked over the error handling code and I liked how readable the parser was to find when errors were being thrown and what the error would look like for the user. My only suggestion for your error handling would be what was mentioned in our discussion to also add some feedback when users put in valid inputs so they can understand that their input went through without error.